File: example-create-article.py
#!/usr/bin/env python3
"""
🚀 Exemple de création d'article via l'API GeeksPatrol
Ce script Python montre comment créer un article avec toutes les options
"""

import requests
import json
from datetime import datetime

# Configuration
API_URL = "http://localhost:4321"  # Changez pour votre URL de production
API_KEY = "6T941FIFyj/zyAzApBD0Qw/hIejqUGxZo2S6GVQXYQ4="  # Votre clé API

def get_categories():
    """Récupère la liste des catégories disponibles"""
    print("📋 Récupération des catégories...")
    
    response = requests.get(
        f"{API_URL}/api/categories/list.json",
        headers={"X-API-Key": API_KEY}
    )
    
    if response.status_code == 200:
        data = response.json()
        print("\nCatégories disponibles:")
        for cat in data['categories']:
            print(f"  - [{cat['id']}] {cat['name']} ({cat['slug']})")
        return data['categories']
    else:
        print(f"❌ Erreur: {response.status_code}")
        return []

def create_article():
    """Crée un article complet avec toutes les options"""
    print("\n✍️  Création de l'article...")
    
    # Données de l'article
    article_data = {
        "title": "Les smartphones pliables de 2025 : Notre comparatif complet",
        "content": """
            <h2>Une nouvelle génération de smartphones pliables</h2>
            <p>2025 marque un tournant pour les smartphones pliables. Les fabricants ont enfin résolu 
            les problèmes de durabilité qui ont longtemps entravé l'adoption de cette technologie.</p>
            
            <h3>Le top 5 des smartphones pliables</h3>
            <ol>
                <li><strong>Samsung Galaxy Z Fold 6</strong> - Le pionnier s'améliore encore</li>
                <li><strong>Google Pixel Fold 2</strong> - L'expérience Android pure</li>
                <li><strong>OnePlus Open 2</strong> - Le meilleur rapport qualité-prix</li>
                <li><strong>Xiaomi Mix Fold 4</strong> - L'innovation chinoise</li>
                <li><strong>Honor Magic V3</strong> - Le plus fin du marché</li>
            </ol>
            
            <h3>Samsung Galaxy Z Fold 6 : Notre choix n°1</h3>
            <p>Samsung continue de dominer le marché avec le Z Fold 6. Cette 6ème génération apporte :</p>
            <ul>
                <li>Écran interne de 7.6 pouces avec 120Hz adaptatif</li>
                <li>Puce Snapdragon 8 Gen 3 optimisée</li>
                <li>Batterie de 4800mAh avec charge 45W</li>
                <li>Certification IPX8 (résistance à l'eau)</li>
                <li>S Pen intégré dans la charnière</li>
            </ul>
            
            <h3>Google Pixel Fold 2 : L'intelligence artificielle au service du pliable</h3>
            <p>Google mise tout sur l'IA avec Gemini intégré directement dans le système. 
            Les fonctionnalités phares incluent :</p>
            <ul>
                <li>Traduction instantanée en mode "livre"</li>
                <li>Retouche photo avancée avec Magic Editor</li>
                <li>Assistant vocal contextuellement aware</li>
                <li>Caméra Tensor G4 avec zoom optique 5x</li>
            </ul>
            
            <blockquote>
                <p>"Les smartphones pliables ne sont plus un gadget. Ils représentent désormais 
                25% des ventes de smartphones premium." - IDC Research, Janvier 2025</p>
            </blockquote>
            
            <h3>Prix et disponibilité</h3>
            <table>
                <thead>
                    <tr>
                        <th>Modèle</th>
                        <th>Prix</th>
                        <th>Disponibilité</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>Samsung Galaxy Z Fold 6</td>
                        <td>1899€</td>
                        <td>Disponible</td>
                    </tr>
                    <tr>
                        <td>Google Pixel Fold 2</td>
                        <td>1799€</td>
                        <td>Pré-commande</td>
                    </tr>
                    <tr>
                        <td>OnePlus Open 2</td>
                        <td>1499€</td>
                        <td>Disponible</td>
                    </tr>
                </tbody>
            </table>
            
            <h3>Notre verdict</h3>
            <p>Si vous cherchez le meilleur smartphone pliable, le Samsung Galaxy Z Fold 6 reste 
            notre recommandation principale pour sa maturité et son écosystème. Le Pixel Fold 2 
            est parfait pour les fans de Google et d'IA, tandis que le OnePlus Open 2 offre le 
            meilleur rapport qualité-prix.</p>
        """,
        "excerpt": "Découvrez notre comparatif complet des meilleurs smartphones pliables de 2025. Samsung, Google, OnePlus : quel modèle choisir ? Prix, caractéristiques et notre verdict.",
        "status": "approved",  # Article publié directement
        "cover_image_path": "https://images.unsplash.com/photo-1598327105666-5b89351aff97?w=1200&h=630",
        "cover_image_alt": "Smartphone pliable Samsung Galaxy Z Fold ouvert",
        "category_ids": [2],  # Catégorie "Mobile"
        "seo_title": "Smartphones Pliables 2025 : Comparatif & Guide d'achat",
        "seo_description": "Comparatif des meilleurs smartphones pliables 2025 : Samsung Z Fold 6, Pixel Fold 2, OnePlus Open 2. Tests, prix, caractéristiques et notre verdict.",
        "seo_keywords": "smartphone pliable, Galaxy Z Fold 6, Pixel Fold 2, OnePlus Open 2, comparatif 2025, téléphone pliant",
        "sources": [
            {
                "name": "Samsung France",
                "url": "https://www.samsung.com/fr/"
            },
            {
                "name": "Google Store",
                "url": "https://store.google.com/"
            },
            {
                "name": "GSMArena",
                "url": "https://www.gsmarena.com"
            },
            {
                "name": "IDC Research",
                "url": "https://www.idc.com"
            }
        ]
    }
    
    # Envoi de la requête
    response = requests.post(
        f"{API_URL}/api/articles/create.json",
        headers={
            "Content-Type": "application/json",
            "X-API-Key": API_KEY
        },
        json=article_data
    )
    
    # Traitement de la réponse
    print(f"\n📊 Statut HTTP: {response.status_code}")
    
    if response.status_code == 201:
        data = response.json()
        article = data['article']
        
        print("\n✅ Article créé avec succès!")
        print(f"   ID: {article['id']}")
        print(f"   Titre: {article['title']}")
        print(f"   Slug: {article['slug']}")
        print(f"   Statut: {article['status']}")
        print(f"   Temps de lecture: {article.get('reading_time', 'N/A')} min")
        
        # Afficher les catégories
        if article.get('article_categories'):
            print(f"   Catégories:")
            for ac in article['article_categories']:
                if ac.get('category'):
                    print(f"     - {ac['category']['name']}")
        
        # Afficher les sources
        if article.get('sources'):
            print(f"   Sources: {len(article['sources'])} source(s)")
        
        # URL de l'article
        category_slug = article['article_categories'][0]['category']['slug'] if article.get('article_categories') else 'blog'
        article_url = f"{API_URL}/{category_slug}/{article['slug']}"
        print(f"\n🌐 URL de l'article: {article_url}")
        
        return article
    else:
        print("\n❌ Erreur lors de la création de l'article")
        try:
            error_data = response.json()
            print(f"   Message: {error_data.get('error', 'Erreur inconnue')}")
            if 'details' in error_data:
                print(f"   Détails: {error_data['details']}")
        except:
            print(f"   Réponse brute: {response.text}")
        return None

def main():
    """Fonction principale"""
    print("=" * 60)
    print("🚀 Création d'article via l'API GeeksPatrol")
    print("=" * 60)
    
    # Récupérer les catégories
    categories = get_categories()
    
    if not categories:
        print("\n❌ Impossible de récupérer les catégories. Vérifiez votre API.")
        return
    
    # Créer l'article
    article = create_article()
    
    if article:
        print("\n✨ Processus terminé avec succès!")
    else:
        print("\n❌ Échec de la création de l'article")
    
    print("=" * 60)

if __name__ == "__main__":
    main()

File: example-create-article.mjs
#!/usr/bin/env node
/**
 * 🚀 Exemple de création d'article via l'API GeeksPatrol
 * Ce script Node.js montre comment créer un article avec toutes les options
 */

const API_URL = 'http://localhost:4321'; // Changez pour votre URL de production
const API_KEY = process.env.API_KEY || 'YOUR_API_KEY_HERE'; // Définir la clé API via variable d'environnement

/**
 * Récupère la liste des catégories disponibles
 */
async function getCategories() {
  console.log('📋 Récupération des catégories...');
  
  try {
    const response = await fetch(`${API_URL}/api/categories/list.json`, {
      headers: {
        'X-API-Key': API_KEY
      }
    });
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    
    const data = await response.json();
    console.log('\nCatégories disponibles:');
    data.categories.forEach(cat => {
      console.log(`  - [${cat.id}] ${cat.name} (${cat.slug})`);
    });
    
    return data.categories;
  } catch (error) {
    console.error('❌ Erreur:', error.message);
    return [];
  }
}

/**
 * Crée un article complet avec toutes les options
 */
async function createArticle() {
  console.log('\n✍️  Création de l\'article...');
  
  // Données de l'article
  const articleData = {
    title: 'Les tendances tech 2025 : Ce qui va révolutionner notre quotidien',
    content: `
      <h2>Un aperçu des innovations majeures de 2025</h2>
      <p>L'année 2025 s'annonce riche en innovations technologiques. Des avancées en IA 
      à la réalité augmentée, découvrez les tendances qui vont transformer notre quotidien.</p>
      
      <h3>1. L'IA générative partout</h3>
      <p>L'intelligence artificielle générative ne se limite plus à ChatGPT. Elle s'intègre 
      désormais dans :</p>
      <ul>
        <li><strong>Les systèmes d'exploitation</strong> : Windows 12 et macOS 15 intègrent des IA natives</li>
        <li><strong>Les applications mobiles</strong> : Génération de contenu en temps réel</li>
        <li><strong>Les appareils photo</strong> : Amélioration intelligente des photos</li>
        <li><strong>Les assistants vocaux</strong> : Conversations naturelles et contextuelles</li>
      </ul>
      
      <h3>2. La réalité augmentée devient mainstream</h3>
      <p>Avec l'arrivée de l'Apple Vision Pro et ses concurrents, la réalité augmentée 
      sort enfin du stade expérimental :</p>
      <ul>
        <li>Shopping virtuel immersif</li>
        <li>Collaboration à distance en 3D</li>
        <li>Éducation interactive et gamifiée</li>
        <li>Tourisme augmenté</li>
      </ul>
      
      <h3>3. Les voitures autonomes niveau 4</h3>
      <p>2025 marque l'arrivée des premières voitures totalement autonomes (niveau 4) 
      sur les routes européennes. Tesla, Mercedes et Waymo ouvrent la voie.</p>
      
      <blockquote>
        <p>"D'ici 2030, 50% des véhicules vendus en Europe auront des capacités de 
        conduite autonome niveau 3 ou supérieur." - McKinsey & Company</p>
      </blockquote>
      
      <h3>4. L'Internet quantique</h3>
      <p>Les premiers réseaux quantiques commerciaux font leur apparition, promettant :</p>
      <ul>
        <li>Sécurité inviolable des communications</li>
        <li>Vitesses de transmission exponentielles</li>
        <li>Nouvelles applications en finance et santé</li>
      </ul>
      
      <h3>5. La 6G en phase de test</h3>
      <p>Alors que la 5G se déploie encore, la 6G entre en phase de test dans plusieurs pays. 
      Les promesses : vitesses 100x supérieures à la 5G et latence quasi-nulle.</p>
      
      <h3>6. Les batteries solides</h3>
      <p>Les premières batteries à électrolyte solide arrivent sur le marché, offrant :</p>
      <ul>
        <li>Autonomie doublée pour les smartphones et véhicules électriques</li>
        <li>Charge complète en 10 minutes</li>
        <li>Durée de vie de 20 ans</li>
        <li>Sécurité accrue (pas de risque d'incendie)</li>
      </ul>
      
      <h3>Impact sur la société</h3>
      <p>Ces technologies vont transformer profondément notre quotidien :</p>
      <ol>
        <li><strong>Travail</strong> : Télétravail immersif et collaboration virtuelle</li>
        <li><strong>Santé</strong> : Diagnostic précoce par IA et chirurgie assistée</li>
        <li><strong>Éducation</strong> : Apprentissage personnalisé et immersif</li>
        <li><strong>Divertissement</strong> : Expériences gaming révolutionnaires</li>
        <li><strong>Mobilité</strong> : Villes intelligentes et transport autonome</li>
      </ol>
      
      <h3>Conclusion</h3>
      <p>2025 ne sera qu'un début. Ces technologies vont continuer à évoluer et converger 
      pour créer un futur que nous commençons à peine à imaginer.</p>
    `,
    excerpt: 'Découvrez les 6 tendances tech majeures de 2025 : IA générative omniprésente, réalité augmentée mainstream, voitures autonomes niveau 4, Internet quantique et bien plus encore.',
    status: 'approved',
    cover_image_path: 'https://images.unsplash.com/photo-1451187580459-43490279c0fa?w=1200&h=630',
    cover_image_alt: 'Technologies futuristes et innovation 2025',
    category_ids: [3], // Catégorie "Tech & Gadgets"
    seo_title: 'Tendances Tech 2025 : Les 6 innovations qui vont tout changer',
    seo_description: 'IA générative, réalité augmentée, voitures autonomes, Internet quantique, 6G et batteries solides : découvrez les tendances tech qui vont révolutionner 2025.',
    seo_keywords: 'tendances tech 2025, innovation technologique, IA générative, réalité augmentée, voitures autonomes, Internet quantique, 6G, batteries solides',
    sources: [
      {
        name: 'McKinsey & Company',
        url: 'https://www.mckinsey.com'
      },
      {
        name: 'MIT Technology Review',
        url: 'https://www.technologyreview.com'
      },
      {
        name: 'Gartner',
        url: 'https://www.gartner.com'
      },
      {
        name: 'TechCrunch',
        url: 'https://techcrunch.com'
      }
    ]
  };
  
  try {
    const response = await fetch(`${API_URL}/api/articles/create.json`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'X-API-Key': API_KEY
      },
      body: JSON.stringify(articleData)
    });
    
    console.log(`\n📊 Statut HTTP: ${response.status}`);
    
    const data = await response.json();
    
    if (response.status === 201) {
      const { article } = data;
      
      console.log('\n✅ Article créé avec succès!');
      console.log(`   ID: ${article.id}`);
      console.log(`   Titre: ${article.title}`);
      console.log(`   Slug: ${article.slug}`);
      console.log(`   Statut: ${article.status}`);
      console.log(`   Temps de lecture: ${article.reading_time || 'N/A'} min`);
      
      // Afficher les catégories
      if (article.article_categories?.length > 0) {
        console.log('   Catégories:');
        article.article_categories.forEach(ac => {
          if (ac.category) {
            console.log(`     - ${ac.category.name}`);
          }
        });
      }
      
      // Afficher les sources
      if (article.sources?.length > 0) {
        console.log(`   Sources: ${article.sources.length} source(s)`);
      }
      
      // URL de l'article
      const categorySlug = article.article_categories?.[0]?.category?.slug || 'blog';
      const articleUrl = `${API_URL}/${categorySlug}/${article.slug}`;
      console.log(`\n🌐 URL de l'article: ${articleUrl}`);
      
      return article;
    } else {
      console.error('\n❌ Erreur lors de la création de l\'article');
      console.error(`   Message: ${data.error || 'Erreur inconnue'}`);
      if (data.details) {
        console.error(`   Détails: ${data.details}`);
      }
      return null;
    }
  } catch (error) {
    console.error('\n❌ Erreur:', error.message);
    return null;
  }
}

/**
 * Fonction principale
 */
async function main() {
  console.log('='.repeat(60));
  console.log('🚀 Création d\'article via l\'API GeeksPatrol');
  console.log('='.repeat(60));
  
  // Récupérer les catégories
  const categories = await getCategories();
  
  if (categories.length === 0) {
    console.log('\n❌ Impossible de récupérer les catégories. Vérifiez votre API.');
    return;
  }
  
  // Créer l'article
  const article = await createArticle();
  
  if (article) {
    console.log('\n✨ Processus terminé avec succès!');
  } else {
    console.log('\n❌ Échec de la création de l\'article');
  }
  
  console.log('='.repeat(60));
}

// Exécuter
main().catch(error => {
  console.error('Erreur fatale:', error);
  process.exit(1);
});
